src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -54,52 +51,106 @@ public ContigAligner(final String referenceFilePath) throws IOException {
+                // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                        .filter(a -> a.getSecondary() < 0)
+                        .map(a -> new AlignmentRegion(breakpointId, contigId, a))
+                        .sorted(Comparator.comparing(a -> a.startInAssembledContig))
+                        .collect(arrayListCollector(alnRgns.length));
+                 alignedContigs.addAll(alignmentRegionList);
             }
         } catch (final IOException e) {
             throw new GATKException("could not execute BWA");
         }
 
-        return assembledBreakpoints;
+        return alignedContigs;
+    }
+
+    @VisibleForTesting
+    public static List<AssembledBreakpoint> getAssembledBreakpointsFromAlignmentRegions(final byte[] sequence, final List<AlignmentRegion> alignmentRegionList, final Integer minAlignLength) {
+        final List<AssembledBreakpoint> results = new ArrayList<>(alignmentRegionList.size() - 1);
+        final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
+        final List<String> insertionAlignmentRegions = new ArrayList<>();
+        if ( iterator.hasNext() ) {
+            AlignmentRegion current = iterator.next();
+            while (treatAlignmentRegionAsInsertion(current) && iterator.hasNext()) {
+                current = iterator.next();
+            }
+            while ( iterator.hasNext() ) {
+                final AlignmentRegion next = iterator.next();
+                if (currentAlignmentRegionIsTooSmall(current, next, minAlignLength)) {
+                    continue;
+                }
+
+                if (treatNextAlignmentRegionInPairAsInsertion(current, next, minAlignLength)) {
+                    if (iterator.hasNext()) {
+                        insertionAlignmentRegions.add(next.toPackedString());
+                        // todo: track alignments of skipped regions for classification as duplications, mei's etc.
+                        continue;
+                    } else {
+                        break;
+                    }
+                }
+
+                final AlignmentRegion previous = current;
+                current = next;
+
+                final byte[] sequenceCopy = Arrays.copyOf(sequence, sequence.length);
+
+                String homology = "";
+                if (previous.endInAssembledContig >= current.startInAssembledContig) {
+                    final byte[] homologyBytes = Arrays.copyOfRange(sequenceCopy, current.startInAssembledContig - 1, previous.endInAssembledContig);
+                    if (previous.referenceInterval.getStart() > current.referenceInterval.getStart()) {
+                        SequenceUtil.reverseComplement(homologyBytes, 0, homologyBytes.length);
+                    }
+                    homology = new String(homologyBytes);
+                }
```

Comment:
this if block can be replaced by a refactor (personal taste--I tend to avoid big `if`, `while`, and `for` blocks. this and the next two make the enclosing `while` block a bit hard to read):

```java
    String homology = getHomology(current, previous, sequenceCopy);

    private static String getHomology(final AlignmentRegion current, final AlignmentRegion previous, final byte[] sequenceCopy) {
        String homology = "";
        if (previous.endInAssembledContig >= current.startInAssembledContig) {
            final byte[] homologyBytes = Arrays.copyOfRange(sequenceCopy, current.startInAssembledContig - 1, previous.endInAssembledContig);
            if (previous.referenceInterval.getStart() > current.referenceInterval.getStart()) {
                SequenceUtil.reverseComplement(homologyBytes, 0, homologyBytes.length);
            }
            homology = new String(homologyBytes);
        }
        return homology;
    }
```